src/services/identity/UsersService.ts

```diff
@@ -137,28 +137,19 @@ class UsersService {
 
   async login(githubId: string): Promise<User> {
     return this.sequelize.transaction<User>(async (transaction) => {
-      // NOTE: The service's findOrCreate is not being used here as this requires an explicit transaction
-      let user = await this.repository.findOne({
+      const [user] = await this.repository.findOrCreate({
         where: { githubId },
         transaction,
       })
-
-      if (!user) {
-        user = await this.repository.create({
-          githubId,
-          transaction,
-        })
-      }
-
       user.lastLoggedIn = new Date()
+
       return user.save({ transaction })
```

Comment:
Hmm, it's probably not necessary to run this save in the transaction 🤔 . The code could be simplified further by removing our custom transaction and use sequlize's internal transaction that is created when calling `findOrCreate()`. Per [documentation](https://sequelize.org/api/v6/class/src/model.js~model#static-method-findOrCreate):

> If no transaction is passed in the options object, a new transaction will be created internally, to prevent the race condition where a matching row is created by another connection after the find but before the insert call.